lib/puppet/file_serving/terminus_helper.rb

```diff
@@ -26,7 +26,8 @@ def path2instances(request, *paths)
     end
 
     Puppet::FileServing::Fileset.merge(*filesets).collect do |file, base_path|
-      path2instance(request, base_path, :ignore_source_permissions => true, :relative_path => file)
+      ignore_source_permissions = (!request.options[:source_permissions] || request.options[:source_permissions] == :ignore)
```

Comment:
The Metadata class also has logic about how to interpret the default `source_permissions` value: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_serving/metadata.rb#L30. I'm thinking we should keep the logic in the Metadata class, and have the terminus_helper treat `source_permissions` like it does `checksum_type`, have the terminus_helper call the `#source_permissions=` method on the model class if specified in `request.options[:source_permissions]`.

I've pushed up a commit https://github.com/joshcooper/puppet/commit/07eb06f7f1517866314fab5459c1abef22959686 to describe what I mean. Amazingly only one test failed (because the test was passing an invalid source_permissions value).